lib/Db/CoreQueryBuilder.php

```diff
@@ -1156,13 +1157,36 @@ public function leftJoinInitiator(
 			)
 		);
 
+
+		$listMembershipCircleAlias = [$aliasMembership];
+		if ($this->getBool('initiatorDirectMember', $options, false)) {
+			try {
+				$aliasDirectInitiator = $this->generateAlias($alias, self::DIRECT_INITIATOR, $options);
+				$listMembershipCircleAlias[] = $aliasDirectInitiator;
+			} catch (RequestBuilderException $e) {
+			}
+		}
+
 		try {
 			$aliasMembershipCircle = $this->generateAlias($aliasMembership, self::CONFIG, $options);
+			$orXMembershipCircle = $expr->orX();
+			array_map(
```

Comment:
if `$listMembershipCircleAlias` can be big, use foreach for it has a much better memory footprint. Otherwise ignore :)

Reply:
it can only contains two fields, the `memberships` table, or the `members` table.